### PR TITLE
GAWB-1276: Send email invitations, update ACL code to handle new responses

### PR DESCRIPTION
**Pull request by MatthewBemis**
Dependencies:

https://github.com/broadinstitute/rawls/pull/572
https://github.com/broadinstitute/thurloe/pull/37
https://github.com/broadinstitute/firecloud-develop/pull/212
https://github.com/broadinstitute/firecloud-develop/pull/213

One deviation from what's described in the parent ticket: instead of saying "FirstName LastName has invited you to...." we say "email@example.com has invited you to..." because the proxy currently doesn't give that information to us so we'd need to change it. We could also look it up in thurloe but......

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/Notification.scala [hunk 1]
```diff
@@ -7,23 +7,23 @@ import org.broadinstitute.dsde.firecloud.FireCloudConfig
...
+  val replyTo: Option[String] = None
   val notificationId: String
   def toMap: Map[String, String]
   def workspacePortalUrl = FireCloudConfig.FireCloud.fireCloudPortalUrl + "/#workspaces/%s:%s"
 }
 
 case class ActivationNotification(recipientUserId: String) extends Notification {
-  val userId = recipientUserId
-  val replyTo = None
+  override val userId = Some(recipientUserId)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/Notification.scala]: Option, not Some

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/Notification.scala [hunk 2]
```diff
@@ -7,23 +7,23 @@ import org.broadinstitute.dsde.firecloud.FireCloudConfig
...
   val notificationId = FireCloudConfig.Notification.activationTemplateId
   def toMap: Map[String, String] = Map.empty
 }
 
 case class WorkspaceAddedNotification(recipientUserId: String, accessLevel: String, workspaceNamespace: String, workspaceName: String, originEmail: String) extends Notification {
-  val userId = recipientUserId
-  val replyTo = Some(originEmail)
+  override val userId = Some(recipientUserId)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/Notification.scala]: Option, not Some

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/Notification.scala [hunk 3]
```diff
@@ -33,12 +33,18 @@ case class WorkspaceAddedNotification(recipientUserId: String, accessLevel: Stri
 }
 
 case class WorkspaceRemovedNotification(recipientUserId: String, accessLevel: String, workspaceNamespace: String, workspaceName: String, originEmail: String) extends Notification {
-  val userId = recipientUserId
-  val replyTo = Some(originEmail)
+  override val userId = Some(recipientUserId)
+  override val replyTo = Some(originEmail)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/Notification.scala]: Option, not Some

**Review by dvoet — DISMISSED** [review]
stop creating Options using Some(xyz), it should be Option(xyz) to avoid ever having hte confusing situation of Some(null)

**Comment by ansingh7115** [comment]
Tests! Otherwise 👍

**Pull request merged**